ledger/pulsemanager/pulsemanager.go

```diff
@@ -258,28 +268,82 @@ func (m *PulseManager) processRecentObjects(
 		pendingRequests[id] = record.SerializeRecord(pendingRecord)
 	}
 
-	dropSizeHistory, err := m.db.GetDropSizeHistory(ctx)
+	dropSizeHistory, err := m.db.GetDropSizeHistory(ctx, jetID)
 	if err != nil {
-		return errors.Wrap(err, "[ processRecentObjects ] Can't GetDropSizeHistory")
+		return nil, errors.Wrap(err, "[ processRecentObjects ] Can't GetDropSizeHistory")
 	}
 
 	msg := &message.HotData{
-		Jet:                *core.NewRecordRef(core.DomainID, jetID),
 		Drop:               *drop,
-		PulseNumber:        previousSlotPulse.Pulse.PulseNumber,
+		PulseNumber:        pulse,
 		RecentObjects:      recentObjects,
 		PendingRequests:    pendingRequests,
 		JetDropSizeHistory: dropSizeHistory,
 	}
-	_, err = m.Bus.Send(ctx, msg, *currentSlotPulse, nil)
+	return msg, nil
+}
+
+func (m *PulseManager) sendExecutorData(
+	ctx context.Context,
+	currentPulse, newPulse *core.Pulse,
+	jetID core.RecordID,
+	msg *message.HotData,
+) error {
+	shouldSplit := func() (bool, error) {
```

Comment:
1. error is not possible here, why function still returns it?
2. I suggest to move split logic in separate function with nice godoc and test